SRG.ipynb

```
@@ -0,0 +1,226 @@
+    "print(f\"Loaded {len(dataset)} graphs.\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "id": "bb12804a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def weisfeiler_lehman_hash(graph):\n",
+    "    colors = {node: data.get('label', 0) for node, data in graph.nodes(data=True)}\n",
```

Ruff (0.12.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)